lib/iris/experimental/ugrid/__init__.py

```diff
@@ -522,6 +522,8 @@ def __eq__(self, other):
             if hasattr(other, "metadata"):
                 # metadata comparison
                 eq = self.metadata == other.metadata
+                if eq:
```

Comment:
This fixes it so that connectivities of different shapes can compare correctly (i.e. returns False).
Without this, I was getting errors from the next code lines due to problems in array comparison.